daemon/cmd/kube_proxy_replacement.go

```diff
@@ -416,10 +416,14 @@ func handleNativeDevices(strict bool) {
 		// NodePort BPF would forward requests to a service endpoint on a remote
 		// node via the Wireguard tunnel device which makes requests to be
 		// encrypted.
-		prevDev := option.Config.DirectRoutingDevice
-		option.Config.DirectRoutingDevice = wireguardTypes.IfaceName
-		log.Infof("Switching direct routing device from %s to %s", prevDev,
-			option.Config.DirectRoutingDevice)
+		if supportL3Dev() {
+			prevDev := option.Config.DirectRoutingDevice
+			option.Config.DirectRoutingDevice = wireguardTypes.IfaceName
+			log.Infof("Switching direct routing device from %s to %s", prevDev,
+				option.Config.DirectRoutingDevice)
+		} else {
+			log.Warn("NodePort BPF forwarded traffic won't be encrypted by Wireguard (requires >= 5.8 kernel)")
```

Comment:
I'm not super comfortable having this as just a warning as it means that contrary to what the user expects some traffic is not going to be encrypted and the only way to find this out is by looking into the logs

Reply:
Agree, shouldn't we fatal instead? It seems odd from one kernel version to another to not encrypt / encrypt.

Reply:
Yeah, makes sense :+1:  Changed.